app/src/components/features/mocksV2/NewFileModal/NewFileTypeSelector.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useCallback } from "react";
+import { ArrowRightOutlined } from "@ant-design/icons";
+import { Button, Card, Col, Row, Typography } from "antd";
+import { AuthConfirmationPopover } from "components/hoc/auth/AuthConfirmationPopover";
+import { useSelector } from "react-redux";
+import { Link, useNavigate } from "react-router-dom";
+import { getUserAuthDetails } from "store/selectors";
+import { redirectToFileMockEditorCreateMock, redirectToMocksList } from "utils/RedirectionUtils";
+import { FileType } from "../types";
+import { IoLogoCss3, IoLogoHtml5, IoLogoJavascript } from "react-icons/io5";
+import { AUTH } from "modules/analytics/events/common/constants";
+
+const FILE_TYPES_CONFIG = {
+  [FileType.HTML]: {
+    TYPE: FileType.HTML,
+    NAME: "Host new HTML file",
+    DESCRIPTION: "Mock an HTML Response",
+    ICON: () => <IoLogoHtml5 />,
+    TOOL_TIP_PLACEMENT: "top",
+  },
+  [FileType.CSS]: {
+    TYPE: FileType.CSS,
+    NAME: "Host new CSS file",
+    DESCRIPTION: "Mock a CSS Response",
+    ICON: () => <IoLogoCss3 />,
+    TOOL_TIP_PLACEMENT: "top",
+  },
+  [FileType.JS]: {
+    TYPE: FileType.JS,
+    NAME: "Host new JS file",
+    DESCRIPTION: "Mock a JS Response",
+    ICON: () => <IoLogoJavascript />,
+    TOOL_TIP_PLACEMENT: "top",
+  },
+};
+
+const NewFileTypeSelector: React.FC<{ handleTypeSelection?: (type: string) => void }> = ({ handleTypeSelection }) => {
+  const navigate = useNavigate();
+  const user = useSelector(getUserAuthDetails);
+
+  const renderFileTypeCard = useCallback(
+    (type: string, data: any) => {
+      return (
```

Comment:
let's use `useMemo`,
```suggestion
const fileTypeCard = useMemo(() => <></>, [])
```